src/Components/Blazor/Build/src/Microsoft.AspNetCore.Blazor.Build.csproj

```diff
@@ -27,10 +28,26 @@
 
   <ItemGroup>
     <ProjectReference Condition="'$(BuildNodeJS)' != 'false' and '$(BuildingInsideVisualStudio)' != 'true'" Include="$(RepoRoot)src\Components\Web.JS\Microsoft.AspNetCore.Components.Web.JS.npmproj" ReferenceOutputAssembly="false" />
-    <Reference Include="Microsoft.Extensions.CommandLineUtils.Sources" />
+    <Reference Include="Microsoft.Build.Framework" />
+    <Reference Include="Microsoft.Build.Utilities.Core" />
     <Reference Include="System.Reflection.Metadata" />
+    <Reference Include="System.Text.Json" />
   </ItemGroup>
 
+  <Target Name="CopyBuildTask" BeforeTargets="Build" AfterTargets="PrepareForRun">
+    <!-- If you see a build warning here and you mean to update the tasks dll when building in VS, manually stop MSBuild.exe processes. -->
+    <Copy SourceFiles="@(IntermediateAssembly)" DestinationFolder="$(OutputPath)tools" SkipUnchangedFiles="true" Retries="1" ContinueOnError="true">
+      <Output TaskParameter="CopiedFiles" ItemName="FileWrites" />
+      <Output TaskParameter="CopiedFiles" ItemName="_CopiedIntermediateAssembly"/>
+    </Copy>
+
+    <Copy SourceFiles="@(ReferenceCopyLocalPaths);@(_DebugSymbolsIntermediatePath)" DestinationFolder="$(OutputPath)tools" SkipUnchangedFiles="true"  Retries="1" ContinueOnError="true">
+      <Output TaskParameter="CopiedFiles" ItemName="FileWrites" />
+    </Copy>
```

Comment:
The code from lines 39-46 is extremely nonobvious to me. Could you add comments saying what these blocks are doing?

Reply:
Oh wait I realise now your comment does attempt to address this. But perhaps your comment in GitHub was clearer than your comment in the code. Could you update the code comment to note how this is a workaround?